consensus/src/lib.rs

```diff
@@ -150,7 +150,7 @@ cfg_if::cfg_if! {
         ///
         /// ## Synchronization
         ///
-        /// The same considerations for `Supervisor` apply here.
+        /// The same considerations for [`Supervisor`](crate::Supervisor) apply here.
```

Comment:
nit: you don't need `(crate::Supervisor)` here since `Supervisor` is in scope, the rustdoc can find it automatically